src/components/ModelizerTextView.vue

```diff
@@ -56,94 +61,159 @@ const props = defineProps({
+    activeFileTab.value = { isSelected: false, id: '' };
   }
+  FileEvent.SelectFileEvent.next(activeFileTab.value);
 }
 
 /**
- * Close file by removing it from files array using its id.
- * If the closed file was the current active file, update activeFile.
- *
- * @param {string} fileId - id of closed file
+ * Close file by removing it from fileTabArray array using its id.
+ * If the closed file was the current active file tab, update activeFileTab.
+ * @param {string} fileId - Id of closed file.
  */
-function closeFile(fileId) {
-  const index = files.value.findIndex((file) => file.id === fileId);
-  if (index !== -1) {
-    files.value.splice(index, 1);
-  }
-  if (fileId === activeFile.value.id) {
-    updateActiveFileId();
+function deleteFileTab(fileId) {
+  fileTabArray.value = fileTabArray.value.filter(({ id }) => id !== fileId);
+  if (fileId === activeFileTab.value.id) {
+    setLastFileActive();
   }
 }
 
 /**
- * Update project nodes and files.
- * If the previous active file's id is not contained in files, update activeFile.
+ * Update project nodes and fileTabArray.
+ * If the previous active file's id is not contained in fileTabArray, update activeFileTab.
  */
 function updateProjectFiles() {
   getProjectFiles(props.projectName)
     .then((fileInformations) => {
-      nodes.value = getTree(props.projectName, fileInformations);
       const projectFilesIds = fileInformations.map((file) => file.path);
-      files.value = files.value.filter(({ id }) => projectFilesIds.includes(id));
-      files.value.forEach((file) => {
+      nodes.value = getTree(props.projectName, fileInformations);
+      fileTabArray.value = fileTabArray.value.filter(({ id }) => projectFilesIds.includes(id));
+
+      fileTabArray.value.forEach((file) => {
         readProjectFile(props.projectName, { path: file.id })
           .then(({ content }) => {
             file.content = content;
           });
       });
-      const isActiveFileInFiles = files.value.find(({ id }) => id === activeFile.value.id);
+
+      const isActiveFileInFiles = fileTabArray.value
+        .find(({ id }) => id === activeFileTab.value.id);
+
       if (!isActiveFileInFiles) {
-        updateActiveFileId();
+        setLastFileActive();
       }
     });
 }
 
-watch(activeFile, () => {
-  FileEvent.SelectFileEvent.next(activeFile.value);
+/**
+ * Set selectedNode to the node param.
+ * @param {Object} node - Node tree Object.
+ */
+function updateSelectedNode(node) {
+  selectedNode.value = node;
+}
+
+/**
+ * Create and add a new file tree, update value of activeFileTab and
+ * send ExpandNode and OpenFile events.
+ * @param {String} fileName - Name of the file to create.
+ */
+function onCreateFileEvent(newFileName) {
+  const newFileId = `${selectedNode.value.id}/${newFileName}`;
+  createFile(newFileId, selectedNode.value.children, newFileName);
+  sortTreeElements([selectedNode.value]);
+  activeFileTab.value = { isSelected: false, id: newFileId };
```

Comment:
put this assignation just under the const line. And leave this one empty.

Reply:
done